internal/pkg/endpoints/kubernetes.go

```diff
@@ -249,6 +249,20 @@ func getPort(o metav1.Object) string {
 	return ""
 }
 
+// parsePath parses a raw URL query, such as `/metrics?format=foo` and returns separately the path and the query.
+// This is needed because the `prometheus.io/path` annotation is often abused, and query arguments are included in it.
+// As HTTP clients will escape out any `?` and `=` present in the path, we need to split it first to form a correct URL.
+func parsePath(pathQuery string) (string, string) {
+	parts := strings.Split(pathQuery, "?")
```

Comment:
could we do something like https://go.dev/play/p/m8JLsUiYEHO?

Reply:
the example is not perfect since I just adapted an existing one

Reply:
Done, I think it's cleaner now :)